web/client-api/src/main/java/io/deephaven/web/client/api/barrage/BarrageUtils.java

```diff
@@ -171,7 +171,8 @@ public static class DeltaUpdatesBuilder {
         private final DeltaUpdates deltaUpdates = new DeltaUpdates();
         private final BarrageUpdateMetadata barrageUpdate;
         private final String[] columnTypes;
-        private int recordBatchesSeen = 0;
+        private long numAddRowsRemaining = 0;
+        private long numModRowsRemaining = 0;
```

Comment:
I might be missing something - but it looks like this value is set once in the constructor; but never decremented? But we have a condition later on it `== 0`? Seems like that will never get hit?

Reply:
You're correct.  When porting, missed a critical line decrementing `numModRowsRemaining`. Please see the change below (line 264)

Reply:
I'm sad we don't have more thorough automated tests for producer -> barrage -> consumer; assuming it would be better at catching something like this. I'll let @nbauernfeind do deeper look into logic.